specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -2,120 +2,148 @@
-the Push payload from it. Care needs to be taken to Get a Deferral and Complete the Deferral while processing the payload for <br/>
-Power Management compliance.
-* If the Activation is Foreground, the app will do 2 things:
-  * It will request a Push Channel Asynchronously with an implementation of Progress and Completed event handler.
-    * Expect Channel operations to take around 2-16 minutes in some rare cases (retryable errors). In the 90th percentile case, <br/>
-    it should be fairly quick operation (in a few seconds).
-  * It will subscribe to a In-memory Push event handler hanging off the retrieved Push Channel component.
-    * The app should ideally handle the event by setting the Handled property to true to prevent Background Activation from <br/>
-    launching a new process.
+
+-   The Registration will take in a CLSID of the COM server as part of the Activator setup. The
+    registration API will simply be a thin wrapper around the BackgroundInfra WinRT APIs and
+    abstract away the COM details from the developer including registration of the inproc Windows
+    App SDK component.
+-   The app will query ApplifeCycle APIs to retrieve an ActivationKind. (Note: This is covered in a
+    seperate AppLifeCycle API spec.)
+-   If the Activation is Push, the app will QI the Arguments property to retrieve an instance of
+    PushActivatedEventArgs and get the Push payload from it. Care needs to be taken to Get a
+    Deferral and Complete the Deferral while processing the payload for Power Management compliance.
+-   If the Activation is Foreground, the app will do 2 things:
+    -   It will request a Push Channel Asynchronously with an implementation of Progress and
+        Completed event handler.
+        -   Expect Channel operations to take around 2-16 minutes in some rare cases (retryable
+            errors). In the 90th percentile case, it should be fairly quick operation (in a few
+            seconds).
+    -   It will subscribe to a In-memory Push event handler hanging off the retrieved Push Channel
+        component.
+        -   The app should ideally handle the event by setting the Handled property to true to
+            prevent Background Activation from launching a new process.
+-   If the application is unpackaged, it will use the ProtocolActivator to register the activator
+    instead. The app will also need to call into the bootstrapper API to have access to the Windows
+    App SDK. Click
+    [here](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp)
+    for information on how to build and deploy unpackaged applications with Windows App SDK.
 
 ```cpp
 int main()
 {
-    // Register the COM Activator GUID
-    PushNotificationActivationInfo info(
-        PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+    PushNotificationActivationInfo info = nullptr;
```

Comment:
NIT: `PushNotificationActivationInfo info{};`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)

Same comment applies throughout